src/category_theory/limits/shapes/regular_mono.lean

```diff
@@ -119,6 +202,56 @@ def regular_epi.desc' {W : C} (f : X ⟶ Y) [regular_epi f] (k : X ⟶ W)
   {l : Y ⟶ W // f ≫ l = k} :=
 cofork.is_colimit.desc' (regular_epi.is_colimit) _ h
 
+/-- If `g` is a regular epi and `h` is a pushout of `g`, then `h` is a regular epi. -/
+def regular_of_is_pushout_snd_of_regular {P Q R S : C} {f : P ⟶ Q} {g : P ⟶ R} {h : Q ⟶ S} {k : R ⟶ S}
+  [gr : regular_epi g] (comm : f ≫ h = g ≫ k) (t : is_colimit (pushout_cocone.mk _ _ comm)) :
+regular_epi h :=
+{ W := gr.W,
+  left := gr.left ≫ f,
+  right := gr.right ≫ f,
+  w := by rw [category.assoc, category.assoc, comm, reassoc_of gr.w],
+  is_colimit :=
+  begin
+    apply cofork.is_colimit.mk' _ _,
+    intro s,
+    have l₁ : gr.left ≫ f ≫ s.π = gr.right ≫ f ≫ s.π,
+      rw [← category.assoc, ← category.assoc, s.condition],
+    obtain ⟨l, hl⟩ := cofork.is_colimit.desc' gr.is_colimit (f ≫ cofork.π s) l₁,
+    obtain ⟨p, hp₁, hp₂⟩ := pushout_cocone.is_colimit.desc' t _ _ hl.symm,
+    refine ⟨p, hp₁, _⟩,
+    intros m w,
+    have z := w.trans hp₁.symm,
+    apply t.hom_ext,
+    apply (pushout_cocone.mk _ _ comm).coequalizer_ext,
+    { exact z },
+    { erw [← cancel_epi g, ← reassoc_of comm, ← reassoc_of comm, z], refl },
+  end }
+
+/-- If `f` is a regular epi and `k` is a pushout of `f`, then `k` is a regular epi. -/
+def regular_of_is_pushout_fst_of_regular {P Q R S : C} {f : P ⟶ Q} {g : P ⟶ R} {h : Q ⟶ S} {k : R ⟶ S}
+  [fr : regular_epi f] (comm : f ≫ h = g ≫ k) (t : is_colimit (pushout_cocone.mk _ _ comm)) :
+regular_epi k :=
```

Comment:
Isn't this just the previous construction with the pushout square flipped over? Surely we shouldn't be reproving it.

Reply:
(Similarly above.)

Reply:
I think it's okay to have these second statements, but they should be proved in closer to one line using the first statements, and the fact that you can flip over a square.

I'd also like to have better names here. I don't know what the `fst` and `snd` you're using in these names refers to.

Reply:
Oh yeah I probably should put in the lemma that flipping a pb is still a pb. Do you have better name suggestions?

Reply:
I think `is_pullback_flip` sounds good.